src/utils/flatten.js

```diff
@@ -52,6 +54,15 @@ export default function flatten(chunk: any, executionContext: ?Object, styleShee
     return `.${chunk.styledComponentId}`;
   }
 
+  /* Warn if not refring styled component */
+  if (process.env.NODE_ENV !== 'production') {
+    if (ReactIs.isElement(chunk)) {
+      return console.warn(
```

Comment:
1. I liked what you were doing before in calling the function and then checking the return to see if it was JSX. That felt more safe to me.

2. Let's not `return`, that'll cause other issues. Just the warning is fine. And let's update the text slightly:

> `${getComponentName(chunk)} is not a styled component and cannot be referred to via component selector. See https://www.styled-components.com/docs/advanced#referring-to-other-components for more details.`

`getComponentName` comes from `src/utils/getComponentName.js`